test/Microsoft.AspNet.Mvc.TagHelpers.Test/LinkTagHelperTest.cs

```diff
@@ -298,12 +287,11 @@ public static TheoryData RunsWhenRequiredAttributesArePresent_Data
             helper.Process(context, output);
 
             // Assert
-            Assert.Null(output.TagName);
-            Assert.NotNull(output.Content);
-            Assert.True(output.IsContentModified);
+            Assert.True(output.IsContentModified || !string.IsNullOrEmpty(output.PostElement.GetContent()));
```

Comment:
Or'd conditions are clear indications you've got more than one test. Also the second condition isn't the best way to check `TagHelperContent`.

Split this test and `RunsWhenRequiredAttributesArePresent_Data` in two. Then `Assert.True(output.IsContentModified); Assert.False(output.PostElement.IsModified)` in one test and the opposite, plus `Assert.False(output.PostElement.IsModified)`, in the other.

If there isn't a clear split or the `IsContentModified` cases overlap the `PostElement.IsModified` ones, divide things up differently but always check both `bool` properties.